### PR TITLE
Suggest different command depending on where 'frozen' config is set

### PR DESCRIPTION
**Pull request by denniss**
Fixes: https://github.com/bundler/bundler/issues/5009

### CODE REVIEW

File: lib/bundler/definition.rb [hunk 1]
```diff
@@ -409,8 +409,10 @@ def ensure_equivalent_gemfile_and_lockfile(explicit_flag = false)
              "updated #{Bundler.default_lockfile.relative_path_from(SharedHelpers.pwd)} to version control."
 
       unless explicit_flag
+
+        suggest_command = Bundler.settings.locations("frozen")[:global] == "1" ? "bundle config --delete frozen" : "bundle install --no-deployment"
```
Comment (segiddins) [inline lib/bundler/definition.rb]: Also, won't we need a `--global` if it is indeed set globally?

Comment (denniss) [inline lib/bundler/definition.rb]: That's one other thing that I saw. I need to re-verify this but it looks like `bundle config --delete frozen` deletes BOTH local and global config.

Comment (denniss) [inline lib/bundler/definition.rb]: May be this is expected?

`➜  bundler-dev cat ~/.bundle/config
---
BUNDLE_DISABLE_CHECKSUM_VALIDATION: "true"
BUNDLE_FROZEN: "1"
➜  bundler-dev cat .bundle/config
---
BUNDLE_PATH: "vendor/bundle"
BUNDLE_DISABLE_SHARED_GEMS: "true"
BUNDLE_FROZEN: "1"
➜  bundler-dev dbundle config --delete frozen
➜  bundler-dev cat ~/.bundle/config
---
BUNDLE_DISABLE_CHECKSUM_VALIDATION: "true"
➜  bundler-dev cat .bundle/config
---
BUNDLE_PATH: "vendor/bundle"
BUNDLE_DISABLE_SHARED_GEMS: "true"
➜  bundler-dev`

Comment (denniss) [inline lib/bundler/definition.rb]: Fixed

Comment (segiddins) [inline lib/bundler/definition.rb]: uh, I'm not entirely sure? @indirect ?

Comment (indirect) [inline lib/bundler/definition.rb]: IIRC, deleting both was our way of punting on the issue. It seems like terrible UI to require specific combinations of multiple flags. So far, it has worked. 😬  We might need to change it if there is an actual usecase that requires deleting one and not the other, though.

Comment (segiddins) [inline lib/bundler/definition.rb]: Cool, just wanted to check

**Comment by denniss** [comment]
@segiddins My bad for force-pushing. `bundle config --delete` appears to delete both local and global config.

```
➜  bundler-dev cat ~/.bundle/config
---
BUNDLE_DISABLE_CHECKSUM_VALIDATION: "true"
BUNDLE_FROZEN: "1"
➜  bundler-dev cat .bundle/config
---
BUNDLE_PATH: "vendor/bundle"
BUNDLE_DISABLE_SHARED_GEMS: "true"
BUNDLE_FROZEN: "1"
➜  bundler-dev dbundle config --delete frozen
➜  bundler-dev cat ~/.bundle/config
---
BUNDLE_DISABLE_CHECKSUM_VALIDATION: "true"
➜  bundler-dev cat .bundle/config
---
BUNDLE_PATH: "vendor/bundle"
BUNDLE_DISABLE_SHARED_GEMS: "true"
```

**Comment by denniss** [comment]
@segiddins I also don't have anything more to add to this one. When you have a moment, can you please take another look?

**Comment by segiddins** [comment]
Excellent!
@bundlerbot r+

**Comment by segiddins** [comment]
@bundlerbot retry

**Comment by trevorturk** [comment]
🤘

**Pull request merged**